src/helpers.rs

```diff
@@ -659,31 +664,49 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
         }
     }
 
+    /// Dereference a pointer operand to a place using `layout` instead of the pointer's declared type
+    fn deref_operand_as(
+        &self,
+        op: &OpTy<'tcx, Provenance>,
+        layout: TyAndLayout<'tcx>,
+    ) -> InterpResult<'tcx, MPlaceTy<'tcx, Provenance>> {
```

Comment:
Now this is entirely equivalent to `deref_operand_and_offset(op, 0, layout, layout)` isn't it?

Reply:
Oh you made `deref_operand_and_offset` call this one. Yeah that also works.